packages/tables/src/Concerns/CanSelectRecords.php

```diff
@@ -45,6 +45,14 @@ public function getAllTableRecordsCount(): int
             return $this->records->total();
         }
 
+        if ($this->isTableSelectionEnabled()) {
+            return $this
+                ->getFilteredTableQuery()
+                ->get()
+                ->reject(fn($item) => is_callable($this->getTableRecordBulkActionsEnabled()) ? !$this->getTableRecordBulkActionsEnabled()($item) : true)
```

Comment:
Filtering on the Collection should only be done if people have a condition for their Bulk Actions.

Reply:
Changed.